pkg/descheduler/strategies/validation/failedpods_strategyparams.go

```diff
@@ -0,0 +1,52 @@
+package validation
```

Comment:
Let's move this under `pkg/descheduler/strategies/failedpods.go` unless @damemi has different plan for the validation code.

Reply:
Yeah, it is a tough one. I mentioned in another comment that I think this validation package will need some work over time but I do agree that perhaps it makes more sense to keep the strategy-specific validation colocated with the strategies themselves :\

Moving out the common code does help clean up a lot of the strategy code, so that much is at least still very helpful. @a7i really sorry to be back and forth on this, but what do you think?

Reply:
I personally like it all in the validation package and move towards a pattern such as below:

```
validation.ValidateStrategyParams(...)
                 .withExcludeOwnerKinds(kinds []string)
                 .withReasons(reasons []string)
                 ...
```

My plan was to refactor the validation package in a subsequent PR. With that being said, in order to move the PR forward, I can go ahead and move it back to the strategy package and revisit in that PR.

Reply:
Please move the code back so we have more time to discuss how we are going to proceed in the next PR.

Reply:
Done